helix-term/src/commands/lsp.rs

```diff
@@ -105,6 +105,23 @@ struct PickerDiagnostic {
     diag: lsp::Diagnostic,
 }
 
+impl PickerDiagnostic {
+    fn severity_marker(&self) -> String {
+        let symbol = self
+            .diag
+            .severity
+            .map(|s| match s {
+                DiagnosticSeverity::HINT => 'H',
+                DiagnosticSeverity::INFORMATION => 'I',
+                DiagnosticSeverity::WARNING => 'W',
+                DiagnosticSeverity::ERROR => 'E',
+                _ => ' ',
+            })
+            .unwrap_or(' ');
+        format!("{} ", symbol)
```

Comment:
I'm a little sad we need to allocate all the time for each and every diagnostic.

If `Span::raw` accept `&'static str`, we could simply embed the space in the returned string and do something like:

```rust
            // Two characters wide, so two spaces in the default
            .map_or("  ", |s| match s {
                DiagnosticSeverity::HINT => "H ",
                DiagnosticSeverity::INFORMATION => "I ",
                DiagnosticSeverity::WARNING => "W ",
                DiagnosticSeverity::ERROR => "E ",
                _ => "  ",
            })
```

Reply:
As long as not config option is offered for this, it would work well

Reply:
A `Span` is a `Cow` so it is possible to create one from a &'static str without allocating so I also thing we should return a string here.. Even if this was configurable it should be possible to use a reference to the config here